cpp/src/binaryop/compiled/util.cpp

```diff
@@ -45,7 +45,11 @@ struct common_type_functor {
         // Eg. d=t-t
         return data_type{type_to_id<TypeCommon>()};
       }
-      return {};
+
+      // A compiler bug may cause a compilation error when using empty initializer list to construct
```

Comment:
This is a specific compiler bug -- I would include as much detail as you can (I suggested specifying this applies to GCC 10 and including a link to a bug report).

Reply:
Do you have any bug report for it? I'll link to if I have one.

Reply:
https://github.com/rapidsai/cudf/pull/10209#discussion_r798896868